examples/rust/ex1-volume/src/main.rs

```diff
@@ -0,0 +1,429 @@
+                    qdata.iter_mut().zip(weights.iter()).enumerate().for_each(
+                        |(i, (qdata, weight))| {
+                            *qdata = (jacobian[i + q * 0] * jacobian[i + q * 3]
+                                - jacobian[i + q * 1] * jacobian[i + q * 2])
+                                * weight
+                        },
+                    );
+                }
+                3 => {
+                    let q = qdata.len();
+                    qdata.iter_mut().zip(weights.iter()).enumerate().for_each(
+                        |(i, (qdata, weight))| {
+                            *qdata = (jacobian[i + q * 0]
+                                * (jacobian[i + q * 4] * jacobian[i + q * 8]
+                                    - jacobian[i + q * 5] * jacobian[i + q * 7])
+                                - jacobian[i + q * 1]
+                                    * (jacobian[i + q * 3] * jacobian[i + q * 8]
+                                        - jacobian[i + q * 5] * jacobian[i + q * 6])
+                                + jacobian[i + q * 2]
+                                    * (jacobian[i + q * 3] * jacobian[i + q * 7]
+                                        - jacobian[i + q * 4] * jacobian[i + q * 6]))
+                                * weight
+                        },
+                    );
+                }
+                _ => unreachable!(),
+            };
+
+            // Return clean error code
+            0
+        };
+    let qf_build_closure = {
+        let mut qfunction = ceed.q_function_interior(1, Box::new(build_mass));
+        qfunction.add_input("dx", (ncomp_x * dim) as i32, EvalMode::Grad);
+        qfunction.add_input("weights", 1, EvalMode::Weight);
+        qfunction.add_output("qdata", 1, EvalMode::None);
+        qfunction
+    };
+// ---------------------------------------------------------------------------------------
+// ISSUE: EvalMode::Weight is not displaying correctly
+println!("{}", qf_build_closure);
+// ---------------------------------------------------------------------------------------
+    // -- QFunction from gallery
+    let qf_build_named = {
+        let name = format!("Mass{}DBuild", dim);
+        ceed.q_function_interior_by_name(&name)
+    };
+    // -- QFunction for use with Operator
+    let qf_build = if gallery {
+        QFunctionOpt::SomeQFunctionByName(&qf_build_named)
+    } else {
+        QFunctionOpt::SomeQFunction(&qf_build_closure)
+    };
+
+    // Operator that build the quadrature data for the mass operator
+    let mut op_build = ceed.operator(qf_build, QFunctionOpt::None, QFunctionOpt::None);
+    op_build.set_field("dx", &restr_mesh, &basis_mesh, VectorOpt::Active);
+    op_build.set_field(
+        "weights",
+        ElemRestrictionOpt::None,
+        &basis_mesh,
+        VectorOpt::None,
+    );
+    op_build.set_field(
+        "qdata",
+        &restr_qdata,
+        BasisOpt::Collocated,
+        VectorOpt::Active,
+    );
```

Comment:
We should make these builders so this non-conditional configuration can be chained.
https://rust-lang.github.io/api-guidelines/type-safety.html#builders-enable-construction-of-complex-values-c-builder

I suppose the examples also suggest this should be `op.field(...)` because `get_` and `set_` prefixes are discouraged in most cases. https://rust-lang.github.io/api-guidelines/naming.html?highlight=get_#getter-names-follow-rust-convention-c-getter

Reply:
yeah, I was noticing that the `get_*` syntax isn't very idiomatic. I can do a pass to remove those in the morning in a separate PR